src/main/java/at/hannibal2/skyhanni/features/nether/SulphurSkitterBox.kt

```diff
@@ -0,0 +1,101 @@
+    private var rodsList = setOf<NEUInternalName>()
+    private var blocksList = setOf<BlockPos>()
+    private val radius = 8
+
+    @SubscribeEvent
+    fun onTick(event: LorenzTickEvent) {
+        if (!isEnabled()) return
+        if (!event.repeatSeconds(1)) return
+        if (InventoryUtils.itemInHandId !in rodsList && config.onlyWithRods) return
+        val from = LocationUtils.playerLocation().add(-20, -20, -20).toBlockPos()
+        val to = LocationUtils.playerLocation().add(20, 20, 20).toBlockPos()
+        blocksList = BlockPos.getAllInBox(from, to).filter {
+            val b = Minecraft.getMinecraft().theWorld.getBlockState(it).block
+            b == Blocks.sponge && it.toLorenzVec().distanceToPlayer() <= 15
+        }.filter {
+            val pos1 = it.add(-radius, -radius, -radius)
+            val pos2 = it.add(radius, radius, radius)
+            BlockPos.getAllInBox(pos1, pos2).any { p ->
+                Minecraft.getMinecraft().theWorld.getBlockState(p).block in listOf(Blocks.lava, Blocks.flowing_lava)
+            }
+        }.toSet()
+    }
+
+    @SubscribeEvent
+    fun onWorldChange(event: LorenzWorldChangeEvent) {
+        blocksList = emptySet()
+    }
+
+    @SubscribeEvent
+    fun onRenderWorld(event: LorenzRenderWorldEvent) {
+        if (!isEnabled()) return
+        if (InventoryUtils.itemInHandId !in rodsList && config.onlyWithRods) return
+        val block = getClosestBlockToPlayer(blocksList) ?: return
+        if (block.toLorenzVec().distanceToPlayer() >= 20) return
+        val pos1 = block.add(-radius, -radius, -radius)
+        val pos2 = block.add(radius, radius, radius)
+        val axis = AxisAlignedBB(pos1, pos2)
+
+        drawBox(axis, event.partialTicks)
+    }
+
+    private fun getClosestBlockToPlayer(list: Set<BlockPos>): BlockPos? {
+        return list.minByOrNull { it.toLorenzVec().distanceToPlayer() }
+    }
+
+    private fun drawBox(axis: AxisAlignedBB, partialTicks: Float) {
+        when (config.boxType) {
+            SulphurSkitterBoxConfig.BoxType.FULL -> {
+                RenderUtils.drawFilledBoundingBox_nea(axis.expandBlock(),
+                    Color(SpecialColour.specialToChromaRGB(config.boxColor), true),
+                    partialTicks = partialTicks,
+                    renderRelativeToCamera = false)
+            }
+
+            SulphurSkitterBoxConfig.BoxType.WIREFRAME -> {
+                RenderUtils.drawWireframeBoundingBox_nea(axis.expandBlock(), Color(SpecialColour.specialToChromaRGB(config.boxColor), true), partialTicks)
+            }
+
+            else -> {
+                RenderUtils.drawWireframeBoundingBox_nea(axis.expandBlock(), Color(SpecialColour.specialToChromaRGB(config.boxColor), true), partialTicks)
+            }
+        }
+    }
+
+    @SubscribeEvent
+    fun onRepoReload(event: RepositoryReloadEvent) {
+        val data = event.getConstant<ItemsJson>("Items")
+        rodsList = data.lava_fishing_rods
```

Comment:
With missing repo data, this set is empty. use something like ?: `emptySet()` as savetly check